gradle-module-plugin/src/main/kotlin/io/ia/sdk/gradle/modl/task/CollectModlDependencies.kt

```diff
@@ -65,12 +69,18 @@ open class CollectModlDependencies @Inject constructor(objects: ObjectFactory, l
 
     @InputFiles
     fun getModlApiDeps(): Configuration {
-        return project.configurations.getByName("modlApi")
+        return project.configurations.getByName(MODULE_API_CONFIGURATION)
     }
 
     @InputFiles
     fun getModlImplementationDeps(): Configuration {
-        return project.configurations.getByName("modlImplementationElements")
+        return project.configurations.getByName(MODULE_IMPLEMENTATION_ELEMENTS)
+    }
+
+    @InputFile
+    fun getJar(): Provider<RegularFile> {
+        // type casting to get at .archiveFile instead of internal-ish DefaultTask.outputs.files
+        return (project.tasks.getByName("jar") as Jar).archiveFile
```

Comment:
I wonder if there's a better way of doing this using public apis?   I see this proposed change as an improvement to what's there, so not a blocker by any means, just have that gut sense that there must be a better way that we've not caught.

Reply:
This is the cleaner, [all-public API](https://docs.gradle.org/7.6/javadoc/org/gradle/api/tasks/bundling/AbstractArchiveTask.html#getArchiveFile--). (And a little simpler too. 😉)

The comment should be more precise. [`DefaultTask.outputs`](https://docs.gradle.org/7.6/javadoc/org/gradle/api/DefaultTask.html#getOutputs--) is public but inexplicably returns `org.gradle.api.internal.TaskOutputsInternal`. At least in 7.6.

Though to be fair, stuff like `.outputs.files` is used all over the place in the Gradleverse. I'm not sure why the return value is internal here.